src/v/container/BUILD

```diff
@@ -0,0 +1,191 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+load("//bazel:test.bzl", "redpanda_cc_bench", "redpanda_cc_btest", "redpanda_cc_gtest")
+
+redpanda_cc_library(
+    name = "intrusive",
+    hdrs = [
+        "include/container/intrusive_list_helpers.h",
+    ],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@boost//:intrusive",
+    ],
+)
+
+redpanda_cc_library(
+    name = "fragmented_vector",
+    hdrs = [
+        "include/container/fragmented_vector.h",
+    ],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+    ],
+)
+
+redpanda_cc_library(
+    name = "chunked_hash_map",
+    hdrs = [
+        "include/container/chunked_hash_map.h",
+    ],
+    strip_include_prefix = "include",
```

Comment:
I wonder if it's worth having macros for new style vs old style of modules. I guess that depends on if we're going to keep the new style with Bazel or not.

Reply:
I have a todo item that is like, "after we remove cmake, move all the headers back to the same directory". wdyt?

Reply:
SGTM

Reply:
https://redpandadata.atlassian.net/browse/CORE-5569